src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -1420,20 +1387,19 @@ private static StatementType GetStatementType(Node astNode)
 
         private static IdentifierNode GetDefinedIdentifier(Node leftNode)
         {
-            if (leftNode is TypedIdentifierNode)
-                return new IdentifierNode(leftNode as IdentifierNode);
-            if (leftNode is IdentifierNode)
+            var lhs = leftNode as TypedIdentifierNode;
+            if (lhs != null)
+                return new IdentifierNode((IdentifierNode) leftNode);
+
+            var identiferNode = leftNode as IdentifierNode;
+            if (identiferNode != null)
             {
-                var identiferNode = leftNode as IdentifierNode;
-                if (identiferNode.ArrayDimensions != null)
-                    return null;
```

Comment:
Identifiers with array dimensions such as `a[0] = x;` were not considered "defined variables." I'm not sure due to what reason. I have removed this restriction so that these list identifiers can also be renamed uniquely in the code block node.